src/meltano/core/tracking/tracker.py

```diff
@@ -194,3 +263,103 @@ def track_command_event(self, event_json: dict[str, Any]) -> None:
                 f"{CLI_EVENT_SCHEMA}/{CLI_EVENT_SCHEMA_VERSION}", event_json
             )
         )
+
+    def track_telemetry_state_change_event(
+        self,
+        setting_name: str,
+        from_value: Union[uuid.UUID, str, bool, None],
+        to_value: Union[uuid.UUID, str, bool, None],
+    ) -> None:
+        """Fire a telemetry state change event.
+
+        Args:
+            setting_name: the name of the setting that is changing
+            from_value: the old value
+            to_value: the new value
+        """
+        logger.debug(
+            f"Telemetry state change detected for '{setting_name}'. "
+            "A one-time 'telemetry_state_change' event will now be sent."
+        )
+        if isinstance(from_value, uuid.UUID):
+            from_value = str(from_value)
+        if isinstance(to_value, uuid.UUID):
+            to_value = str(to_value)
+
+        self.track_unstruct_event(
+            SelfDescribingJson(
+                (
+                    TELEMETRY_STATE_CHANGE_EVENT_SCHEMA
+                    + "/"
+                    + TELEMETRY_STATE_CHANGE_EVENT_SCHEMA_VERSION
+                ),
+                {
+                    "setting_name": setting_name,
+                    "changed_from": from_value,
+                    "changed_to": to_value,
+                },
+            )
+        )
+
+    @property
+    def _analytics_json_path(self) -> Path:
+        """Return path to the 'analytics.json' file."""
+        return self.project.meltano_dir() / "analytics.json"
+
+    @property
+    def _analytics_json_settings(self) -> AnalyticsSettings:
```

Comment:
IMO it's a bit weird to have a "private" computed property. What do you think of making this a proper method since it's only called once anyway?

@WillDaSilva or @pandemicsyn might have opinions too 🙂

Reply:
+1 I'd prefer this one be a method, IMHO this does a lot for being a "property" .

Reply:
Agreed - this one grew and morphed quite a lot. I tidied it and moved the helper function out to its own private method.